packages/protocol/contracts/thirdparty/LibBytesUtils.sol

```diff
@@ -112,16 +112,17 @@ library LibBytesUtils {
     }
 
     function toBytes32(bytes memory _bytes) internal pure returns (bytes32) {
-        if (_bytes.length < 32) {
+        if (_bytes.length == 0) {
+            return 0;
+        } else if (_bytes.length < 32) {
             bytes32 ret;
             assembly {
                 ret := mload(add(_bytes, 32))
             }
-            return ret;
+            return ret >> (8 * (32 - _bytes.length));
```

Comment:
This is problematic:

`ret := mload(add(_bytes, 32)`

because it can overread the _bytes if less than 32 bytes.

Proposed solution:
```
function toBytes32(bytes memory _bytes) internal pure returns (bytes32) {
        if (_bytes.length < 32) {
            return bytes32(_bytes); // It will not over-read
        }
        return abi.decode(_bytes, (bytes32)); // will truncate if input length >32 bytes
            // 32 bytes
    }
```

Kirk also confirmed this one.